test/mpdecimal/nif_test.exs

```diff
@@ -0,0 +1,23 @@
+defmodule MPDecimal.NifTest do
```

Comment:
I pulled these unit tests into a separate file to make it clear which things are testing the `Nif` module and which ones are testing the public interface.

Reply:
Because we made the choice to rely on Elixir code to provide null terminated strings, we'll get undefined behavior if we pass non-null terminated strings into MPDecimal.Nif.power() I can't think of a good way to test this, but I had considered adding a check for properly formed cstrings in the C code and raising an error if the null terminator is not found where expected.

Reply:
If that check is added, then it would be easy to write a test here that exercises that particular programming error to validate that the check is working properly. This would get us closer to adhering to the principle of least surprise.

Reply:
🤔 great point on this. I had forgotten about expecting these to be null-terminated on the elixir side.

The main intent of these tests for me was to validate that there was no easy way to accidentally crash the VM by passing invalid arguments to this function. I *think* we have validated that improperly terminated strings do not crash the VM, they just cause parsing errors on the c-side of the nif and we end up getting `NaN` results. Is that right? I'm not too worried about doing the extra validation since we are clearly documenting that we expect people to use the `MPDecimal.power/2` function, and we are not documenting this interface, so I don't feel too strongly about adding extra validation on this.